### ISSUE TITLE
Sync: Large settings cause a retry loop

### ISSUE THREAD

**Issue by connor4312** [microsoft/vscode#86178]
#85944

If the user settings are large, above the 100KB limit we have set for the server right now, sync fails 'silently' and keeps retrying, as seen in the log data. We should display errors that require user intervention.

```
[2019-12-03 13:51:46.307] [userDataSync] [error] Error: Server returned 413
    at d.write (file:///Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:401:502)
    at async e.apply (file:///Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:391:681)
    at async e.sync (file:///Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:389:844)
    at async d.sync (file:///Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:399:154)
    at async u.sync (file:///Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:395:816)
```

**Comment by sandy081** [comment]
this is a new limitation that got introduced recently. @Tyriar FYI

**Comment by Tyriar** [comment]
Whatever amount we decide on the payload should be measured before it's sent over so you don't need to hit the server to know it's going to be a 413.

**Comment by sandy081** [comment]
I do not think hardcoding the limit on the client is good idea. It is not expected to change VS Code and release whenever we want to change the limit.

**Comment by Tyriar** [comment]
@sandy081 👍, 413 should be handled explicitly in the client though.

**Comment by sandy081** [comment]
I think client still need to know the max size limit from the server. It would be useful if server returns the max limit size along with 413 status code.

**Comment by sandy081** [comment]
Following error is reported to the user in this case and Sync is turned off to prevent further requests to server.

<img width="452" alt="image" src="https://user-images.githubusercontent.com/10746682/73102103-6faef980-3ef1-11ea-8f81-e66dd804abc7.png">

**Comment by sandy081** [comment]
Fixed by a27c9fc6b18f5235aa0a457f5deae69e77499e47
